client/src/components/app.jsx

```diff
@@ -43,36 +45,39 @@ class ProductFeatures extends Component {
     });
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     const productId = window.location.pathname.split('/')[1] || 1000;
-    console.log('productId:', productId);
-    this.getProductFeatures(productId);
-    this.getProductPhotos(productId); // pending service availability
+    await this.getProductFeatures(productId);
+    // this.getProductPhotos(productId); // pending service availability
+    this.setState({isLoading: false});
   }
 
   render() {
     const text = this.state.productFeatures[0];
     const img = this.state.productPhotos;
     return (
-      (text === undefined
-        ? <h1 style={{textAlign: 'center'}}>Product Features Service Unavailable 404</h1>
-        : <div>
-            <GlobalStyles />
-            <Banner
-              bannerText={text.banner}
-              bannerImage={img} />
-            <Features
-              featuresText={text.features}
-              featuresPhotos={img} />
-            <Setup
-              featuresText={text.featureSetup}
-              featuresSetupPhotos={img} />
-            <AdditionalFeatures
-              additionalFeaturesText={text.additionalFeatures}
-              additionalFeaturesPhotos={img} />
-          </div>
+      (this.state.isLoading
```

Comment:
I like how you're showing something specific when loading instead of just having empty things for a second or two.